### PR TITLE
Scripterror

### PR DESCRIPTION
**Pull request by MaxShehovtsov**
1) Removed ai.0.js from Script error's stack trace
2) Added URL custom property showing document.url to make the error more friendly

### CODE REVIEW

**Comment by MaxShehovtsov** [comment]
@olegsych  this commit addresses issue with Util.dump() for errors https://github.com/Microsoft/ApplicationInsights-JS/commit/30eb7c235a22527ec9ebc8fddf9386d46d7cb374

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts [hunk 1]
```diff
@@ -697,6 +718,38 @@ class AppInsightsTests extends TestClass {
             }
         });
 
+        this.testCase({
+            name: "AppInsights._onerror add document URL in case of CORS error",
+            test: () => {
+                // prepare
+                var sut = new Microsoft.ApplicationInsights.AppInsights(this.getAppInsightsSnippet());
```
Comment (ameyagholkar) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: `sut`? :laughing:

File: JavaScript/JavaScriptSDK/Util.ts [hunk 2]
```diff
@@ -155,6 +155,10 @@
         public static dump(object: any): string {
             var objectTypeDump: string = Object.prototype.toString.call(object);
             var propertyValueDump: string = JSON.stringify(object);
+            if (objectTypeDump === "[object Error]") {
+                propertyValueDump = "{ stack: '" + object.stack + "', message: '" + object.message + "', name: '" + object.name + "'";
```
Comment (ameyagholkar) [inline JavaScript/JavaScriptSDK/Util.ts]: :+1:

Comment (ameyagholkar) [inline JavaScript/JavaScriptSDK/Util.ts]: I don't think `object.stack` will work for all browsers. Chrome's Error does not have that property on its prototype. I would much rather loop over properties and dump them into a new json.

Comment (ameyagholkar) [inline JavaScript/JavaScriptSDK/Util.ts]: Nvm. I see that you are adding the `stack` below. Maybe add that as a note before line 159?

**Comment by ameyagholkar** [comment]
Only aesthetic comments. So, :ship: it, :sheep: it!

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts (hunk 1 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: sut! Took it from Oleg's tests and wanted to ask him what that mean. But his authority made me stop and google it first, and it gave me "system under test". :)

File: JavaScript/JavaScriptSDK/Util.ts (hunk 2 above)
Comment (MaxShehovtsov) [inline JavaScript/JavaScriptSDK/Util.ts]: It's going to give me "undefined" in this case I suppose.

File: JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts (hunk 1 above)
Comment (olegsych) [inline JavaScript/JavaScriptSDK.Tests/CheckinTests/appInsights.tests.ts]: Yeah, I wish I could claim I invented that one :smile:

**Pull request merged**